app/services/support_interface/provider_access_controls_export.rb

```diff
@@ -0,0 +1,96 @@
+module SupportInterface
+  class ProviderAccessControlsExport
+    def data_for_export
+      providers = Provider.all
+
+      providers.map do |provider|
+        {
+          name: provider.name,
+          dsa_signer: get_dsa_signer_email(provider),
+          last_user_permissions_change_at: user_permissions_last_changed_at(provider),
+          total_user_permissions_changes: total_user_permissions_changes(provider),
+          user_permissions_changed_by: user_permissions_changed_by(provider),
+          total_manage_users_users: total_manage_users_users(provider),
+          total_manage_orgs_users: total_manage_orgs_users(provider),
+          total_users: provider.provider_users.count,
+          last_org_permissions_change_at: org_permissions_last_changed_at(provider),
+          total_org_permissions_changes: total_org_permissions_changes(provider),
+          org_permissions_changed_by: org_permissions_changed_by(provider),
+          total_org_relationships_as_trainer: total_org_relationships_as_trainer(provider),
+          total_org_relationships: total_org_relationships(provider),
+        }
+      end
+    end
+
+  private
+
+    def get_dsa_signer_email(provider)
+      provider
+        .provider_agreements
+        .data_sharing_agreements
+        .where.not(accepted_at: nil)
+        .order(accepted_at: :desc)
+        .first
+        .provider_user
```

Comment:
I think this will blow up in production because there are providers without signed DSAs. I wonder if we should scope this export to only cover those with signed DSAs?

Reply:
Ah, I see fixed in following commit! On reflection I think it's right to do as you do and leave in all providers, not just DSA signers, as some may be on one side of a relationship with another.